utils/doclint/generateDotnetApi.js

```diff
@@ -0,0 +1,695 @@
+      if (translatedArguments.includes(null))
+        throw 'Unexpected null in translated argument types. Aborting.';
+
+      let argDocumentation = XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth);
+      for (const newArg of translatedArguments) {
+        const sanitizedArgName = newArg.match(/(?<=^[\s"']*)(\w+)/g, '')[0] || newArg;
+        const newArgName = `${argName}${sanitizedArgName[0].toUpperCase() + sanitizedArgName.substring(1)}`;
+        pushArg(newArg, newArgName, arg);
+        addParamsDoc(newArgName, argDocumentation);
+      }
+      return;
+    }
+
+    addParamsDoc(argName, XmlDoc.renderTextOnly(arg.spec, maxDocumentationColumnWidth));
+
+    if (argName === 'timeout' && argType === 'decimal') {
+      args.push(`int timeout = 0`); // a special argument, we ignore our convention
+      return;
+    }
+
+    pushArg(argType, argName, arg);
+  };
+
+  member.args.forEach(parseArg);
+
+  output(XmlDoc.renderXmlDoc(member.spec, maxDocumentationColumnWidth).map(x => `\t${x}`));
+  paramDocs.forEach((val, ind) => {
+    if (val && val.length === 1)
+      output(`/// <param name="${ind}">${val}</param>`);
+    else {
+      output(`/// <param name="${ind}">`);
+      output(val.map(l => `\t/// ${l}`));
+      output(`/// </param>`);
+    }
+  });
+  output(`${type} ${name}(${args.join(', ')});`);
+}
+
+/**
+ * 
+ *  @callback generateNameCallback
+ *  @param {Documentation.Type} t
+ *  @returns {string}
+ */
+
+/**
+ *  @param {Documentation.Type} type 
+ *  @param {Documentation.Class|Documentation.Type} parent
+ *  @param {generateNameCallback} generateNameCallback
+*/
+function translateType(type, parent, generateNameCallback = null) {
+
+  if (!generateNameCallback) {
+    generateNameCallback = (t) => {
+      return t.name;
+    }
+  }
+
+  // a few special cases we can fix automatically
+  if (type.expression === '[null]|[Error]')
+    return 'void';
+  else if (type.expression === '[boolean]|"mixed"')
+    return 'MixedState';
+
+  let isNullableEnum = false;
+  if (type.union) {
+    if (type.union[0].name === 'null') {
+      // for dotnet, this is a nullable type
+      // if the other side is a primitive type
+      if (type.union.length > 2) {
+        if (type.union.filter(x => x.name.startsWith('"')).length == type.union.length - 1)
+          isNullableEnum = true;
+        else
+          throw `Union (${parent.name}) with null is too long.`;
+      } else {
+        const innerTypeName = translateType(type.union[1], parent, generateNameCallback);
+        // if type is primitive, or an enum, then it's nullable
+        if (innerTypeName === 'bool'
+          || innerTypeName === 'int') {
+          return `${innerTypeName}?`;
+        }
+
+        // if it's not a value type, it'll be nullable by default, so we can ignore it
+        return `${innerTypeName}`;
+      }
+    }
+
+    if (type.union.filter(u => u.name.startsWith(`"`)).length == type.union.length
```

Comment:
You can detect both nullable and non-nullable union by looking at `type.unionName` which is always not null for the head element or for the second element for nullable unions (see https://github.com/microsoft/playwright/blob/96e099aca32f87f2d88c1cdb8acbddfae645b7c7/utils/doclint/documentation.js#L432)